1-Draft/RFC00XX-PredIctive-IntelliSense.md

```diff
@@ -0,0 +1,346 @@
+F2            | Profile Code           | Switches between InLineView and ListView
+Ctrl-b	      | Profile Code	         | Add command to history
+Ctrl-q	      | TabCompleteNext	       | Windows style Tab for Emacs mode
+Ctrl-Q	      | TabCompletePrevious	   | Windows style Tab for Emacs mode
+Ctrl-c	      | Copy	                 | Clipboard interaction for Emacs mode
+Ctrl-v	      | Paste	                 | Clipboard interaction for Emacs mode
+Ctrl-d,Ctrl-c | CaptureScreen	         | Captures screen for pasting to email or blog
+Alt-d	        | ShellKillWord	         | Token based movement bound to Emacs word movement
+Alt-Backspace | ShellBackwardKillWord  | “
+Alt-b	        | ShellBackwardWord	     | “
+Alt-f	        | ShellForwardWord	     | “
+Alt-B	        | SelectShellBackwardWord| “
+Alt-F	        | SelectShellForwardWord | “
+“ or ‘	      | Profile Code	         | Insert Smart quotes
+( or { or [	  | Profile Code	         | Insert matching braces
+Backspace	    | Profile Code    	     | Delete previous character
+Alt-w	        | Profile Code	         | Save current line in history – don’t execute
+Ctrl-V	      | Profile Code    	     | Paste clipboard as a Here string
+Alt-(	        | Profile Code	         | Insert parenthesis around selection or entire line
+Alt-‘	        | Profile Code	         | Toggle quotes on argument under cursor
+Alt-%	        | Profile Code	         | Replace aliases with resolved commands
+
+### Search history based on cursor location
+
+The history from PSReadLine is searchable when the user begins typing into the shell.
+Users may search the entire prediction history for commands matching under the current cursor
+position. Users may search history based on 'Verb', 'Noun' and 'Parameter' of a cmdlet
+using the up/down arrow keys.
+
+- Searching history from cmdlet verb.
+![Verb](./media/pi-arrow-verb.gif)
+
+- Searching history from cmdlet noun.
+![noun](./media/pi-arrow-noun.gif)
+
+- Searching history from parameter.
+![Param](./media/pi-arrow-param.gif)
+
+### Edit predictions before accepting the prediction
+
+Users may navigate prediction tokens to make changes to parameters, arguments, and additional
+pipeline commands. The prediction model will update the displayed prediction from the closest
+matches in PSReadLine history or added provider.
+
+In the example below, a user navigates the prediction, pausing to change the -Property argument and
+receiving an updated prediction.
+
+![arg change](./media/pi-fb.gif)
+
+## Prediction Extension Model
+
+Additional providers enhance historic predictions by providing domain specific commands and task
+completions. These providers register with the PowerShell engine which includes an extension model
+to support the registration.
+
+## Extension Model Goals/Non-Goals
+
+ Goals:
+
+- Provide extension model to support additional providers
+- Support the discovery of providers using `Get-PSSubsystem`
+- Provide Module owners method of registration
```

Comment:
Should specify the changes requested by AzPredictor in terms of providing feedback back to the predictor.  We also need public docs on writing a predictor.

Reply:
Not sure I understand your point - are you referring to the iCommand interface feedback or the survey feedback mechanism?

On docs - I started talking to @daxian-dbw about this - I'll follow up and get something started.

Reply:
306 - improvements to the predictors through feedback.